reflex/utils/exec.py

```diff
@@ -29,18 +73,38 @@ def run_process_and_launch_url(
     Args:
         run_command: The command to run.
     """
-    process = processes.new_process(
-        run_command,
-        cwd=constants.WEB_DIR,
-    )
-
-    if process.stdout:
-        for line in process.stdout:
-            if "ready started server on" in line:
-                url = line.split("url: ")[-1].strip()
-                console.print(f"App running at: [bold green]{url}")
-            else:
-                console.debug(line)
+    json_file_path = os.path.join(constants.WEB_DIR, "package.json")
+    last_hash = detect_package_change(json_file_path)
+    process = None
+    first_run = True
+
+    while True:
+        if process is None:
+            process = processes.new_process(
+                run_command,
+                cwd=constants.WEB_DIR,
+            )
+        if process.stdout:
+            for line in process.stdout:
+                if "ready started server on" in line:
+                    if first_run:
+                        url = line.split("url: ")[-1].strip()
+                        console.print(f"App running at: [bold green]{url}")
+                        first_run = False
+                    else:
+                        console.print(f"New packages detected updating app...")
+                else:
+                    console.debug(line)
+                    new_hash = detect_package_change(json_file_path)
+                    if new_hash != last_hash:
+                        last_hash = new_hash
+                        kill(process.pid)
+                        process = None
+                        break
+        if process is not None:
+            break
+
+    print("App stopped")
```

Comment:
we already log `[11:48:49] Reflex app stopped.` in `console.py:81`; so is this needed?